### PR TITLE
Multi label support

### PR DESCRIPTION
**Pull request by netheril96**
This is an attempt to clean up and improve upon @sguada work in #523. It is a work in progress, not ready for merge yet.

### CODE REVIEW

**Comment by elaksana** [comment]
Same here, waiting for this to be merged.

File: src/caffe/layers/multi_label_loss_layer.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const Dtype* target = bottom[1]->cpu_data();
+  Dtype loss = 0;
+  for (int i = 0; i < count; ++i) {
+    if (target[i] != 0) {
+    // Update the loss only if target[i] is not 0
+      loss -= input_data[i] * ((target[i] > 0) - (input_data[i] >= 0)) -
+          log(1 + exp(input_data[i]
+                      - 2 * input_data[i] * (input_data[i] >= 0)));
+    }
+    if (top.size() >= 1) {
```
Comment (immars) [inline src/caffe/layers/multi_label_loss_layer.cpp]: should this assignment outside for loop?

**Comment by immars** [comment]
@netheril96 can you provide a example net definition showing how to use multi-label loss layers?
As Datum.label is not `repeated`, is it correct that I introduce a separate data layer for label vectors?

**Comment by misaghkord** [comment]
when i try to built this pr i get the following error
make: *** [.build_release/src/caffe/layers/multi_label_accuracy_layer.o] Error 1
can anyone help me please?

**Comment by danielhauagge** [comment]
@netheril96, has this been abandoned?

**Comment by danielhauagge** [comment]
@netheril96, has this been abandoned?

**Comment by bhack** [comment]
@danielhauagge I think that core developers want to declare this solved with https://github.com/BVLC/caffe/issues/1698. So I think that will be hard to have multi label support from text file like with this.

**Comment by sayanghosh** [comment]
I have succesfully implemented multilabel support along with a soft max loss function. Will plan to subit a PR after ICCV deadline. I don't know why multi-label doesn't have support, it being a potentially important fetaure. @netheril96  @bhack

**Comment by netheril96** [comment]
@danielhamngren 

Yes, I abandoned this PR.

**Comment by gavinmh** [comment]
@sayanghosh can you share your work on multilabel support?
@bhack can you explain how this feature is considered complete and why this PR is being abandoned?
